bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/discovery/LiquidCheckDiscoveryService.java

```diff
@@ -0,0 +1,211 @@
+    private HttpClient httpClient;
+
+    @Activate
+    public LiquidCheckDiscoveryService(@Reference HttpClientFactory httpClientFactory) {
+        super(SUPPORTED_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS, false);
+        httpClient = httpClientFactory.getCommonHttpClient();
+    }
+
+    /**
+     * Method for starting the scan
+     */
+    @Override
+    protected void startScan() {
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            logger.debug("Couldn't start client: {}", e.getMessage());
+        }
+        scheduler.execute(liquidCheckDiscoveryRunnable());
+    }
+
+    /**
+     * Method to stop the scan
+     */
+    @Override
+    protected synchronized void stopScan() {
+        try {
+            httpClient.stop();
+        } catch (Exception e) {
+            logger.debug("Couldn't stop client: {}", e.getMessage());
+        }
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    /**
+     * Method for creating a Runnable to start a scan
+     * 
+     * @return the Runnable
+     */
+    protected Runnable liquidCheckDiscoveryRunnable() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    List<InetAddress> addresses = getIPv4Adresses();
+                    List<InetAddress> hosts = findActiveHosts(addresses);
+                    for (InetAddress host : hosts) {
+                        Request request = httpClient.newRequest("http://" + host.getHostAddress() + "/infos.json")
+                                .method(HttpMethod.GET).followRedirects(false);
+                        try {
+                            ContentResponse response = request.send();
+                            if (response.getStatus() == 200) {
+                                CommData json = null;
+                                try {
+                                    json = new Gson().fromJson(response.getContentAsString(), CommData.class);
+                                } catch (JsonSyntaxException e) {
+                                    logger.debug("Json Syntax Exception!");
+                                }
+                                if (null != json) {
+                                    if (InetAddress.getByName(json.payload.wifi.station.hostname).isReachable(50)) {
+                                        buildDiscoveryResult(json, true);
+                                    } else {
+                                        buildDiscoveryResult(json, false);
+                                    }
+                                } else {
+                                    logger.debug("Response Object is null!");
+                                }
+                            }
+                        } catch (TimeoutException e) {
+                            logger.debug("TimeOut!");
```

Comment:
Consider something like this to make the log more useful:
```suggestion
                            logger.debug("Timeout: {}", e.getMessage());
```